### PR TITLE
Add base Table component from BE to UI Kit w/ Skeleton Loader

### LINKED ISSUE #10: Move BlockExplorer's Table to ui-kit
[component](https://github.com/casper-network/casper-blockexplorer-frontend/tree/dev/app/src/components/base/Table)

**This tasks is more complex then average reusable component**

### LINKED ISSUE #56: Move the skeleton loader to the UI Kit
we implemented a skeleton loader in the block explorer, but that component should be int the ui-kit

### PR DESCRIPTION
**Pull request by DorianKwan**
### 🔥 Summary
Closes #10 
Closes #56 

Add base Table component from BE to UI Kit with skeleton loader

### 😤 Problem / Goals
- Refactor Table component from BE to be a re-usable ui it component

### 🤓 Solution
- Converted all theming styles to props to allow end users to theme it however they want

### 🗒️ Additional Notes
- Skeleton loader isn't worth exporting from this package as it would be redundant.  However, the table component comes with the skeleton loader from this kit
- A new ticket was created to write tests for this component
- Storybook story for this component is very simple as a highly complex story was taking ages (will we attack at a later date)

New Ticket:  #139

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "cSpell.words": ["CSPR"]
```
Comment (mgibson-casper) [inline .vscode/settings.json:2]: What's this `c.Spell`?

Comment (DorianKwan) [reply .vscode/settings.json:2]: It's for the spell checker VS Code plugin
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

**Review by mgibson-casper — APPROVED** [review]
Nice!

File: .vscode/settings.json (hunk 1 above)
Comment (mgibson-casper) [reply .vscode/settings.json:2]: Oh sweet, good to know!

**Review by Thistle-Strum — APPROVED** [review]

**Pull request merged**